xa/java/jms/department-helidon-teq/src/main/java/com/oracle/mtm/sample/Configuration.java

```diff
@@ -0,0 +1,166 @@
+import oracle.tmm.common.TrmConfig;
+import oracle.tmm.jta.common.DataSourceInfo;
+import oracle.ucp.jdbc.PoolDataSource;
+import oracle.ucp.jdbc.PoolDataSourceFactory;
+import oracle.ucp.jdbc.PoolXADataSource;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.context.Initialized;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import jakarta.jms.JMSException;
+import jakarta.jms.Session;
+import jakarta.jms.Topic;
+import jakarta.jms.TopicConnection;
+import jakarta.jms.TopicSession;
+import jakarta.jms.XATopicConnectionFactory;
+import java.lang.invoke.MethodHandles;
+import java.sql.SQLException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@ApplicationScoped
+public class Configuration {
+
+    private PoolXADataSource xaDataSource;
+    private PoolDataSource dataSource;
+    private Logger logger;
+
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.url")
+    String url;
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.user")
+    String user;
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.password")
+    String password;
+
+    @Inject
+    @ConfigProperty(name = "departmentDataSource.rmid")
+    String rmid;
+
+    private void init(@Observes @Initialized(ApplicationScoped.class) Object event) {
+        initialiseTrmLogger();
+        initialiseDataSource();
+        new Thread(() -> {
+            consumeEvents();
+        }).start();
+
+    }
+
+    private void consumeEvents() {
+        // create a JMS topic connection and session
+        XATopicConnectionFactory tcf = null;
+        try {
+            tcf = AQjmsFactory.getXATopicConnectionFactory(this.getXADatasource());
+            TopicConnection conn = tcf.createXATopicConnection();
+            conn.start();
+            TopicSession session =
+                    (AQjmsSession) conn.createSession(true, Session.AUTO_ACKNOWLEDGE);
+
+            // create a subscriber on the topic
+            Topic topic = ((AQjmsSession) session).getTopic(user, "microtxshqueue2");
+            AQjmsTopicSubscriber subscriber =
+                    (AQjmsTopicSubscriber) session.createDurableSubscriber(topic, "my_sub");
+
+            System.out.println("Waiting for messages...");
+
+            // wait forever for messages to arrive and print them out
+            int i = 0;
+            while (true) {
+
+                // the 1_000 is a one second timeout
+                AQjmsTextMessage message = (AQjmsTextMessage) subscriber.receive(1000);
+                System.out.println("Checkign for message " + i++);
+                if (message != null) {
+                    if (message.getText() != null) {
+                        System.out.println("Message consumed:" + message.getText());
+                    } else {
+                        System.out.println();
```

Comment:
I think we may not need this else condition or else logs will have empty lines every second.
we can comment this println, if required developer can uncomment and see

Reply:
Make sense. Removed the extra newline print statement